src/icespeak/voices/azure.py

```diff
@@ -179,10 +179,16 @@ def load_api_keys(self):
         AzureVoice.AZURE_REGION = API_KEYS.azure.region.get_secret_value()
 
     @override
-    def text_to_speech(self, text: str, options: TTSOptions):
-        speech_conf = speechsdk.SpeechConfig(
-            subscription=AzureVoice.AZURE_KEY, region=AzureVoice.AZURE_REGION
-        )
+    def text_to_speech(self, text: str, options: TTSOptions, keys_override: Keys | None = None):
+        if keys_override and keys_override.azure:
+            _LOG.info(f"Using overridden Azure keys")
+            subscription = keys_override.azure.key
+            region = keys_override.azure.region
+        else:
+            _LOG.info(f"Using default Azure keys")
```

Comment:
Ditto for these log messages, f-string to normal string and debug instead of info.